### DISCUSSION TITLE
Remember size (and position) of full window between launches

### DISCUSSION THREAD

**Discussion by D4n2021** [File-New-Project/EarTrumpet#752] [Ideas]
When opening, it looks like this:

![grafik](https://user-images.githubusercontent.com/20766261/103290079-fcf3ab80-49e8-11eb-96c8-b378402a587b.png)

Then I manually resize it to this:

![grafik](https://user-images.githubusercontent.com/20766261/103290103-0bda5e00-49e9-11eb-8840-1ac434ff59c5.png)

When then closing the window and opening again, it's reset back to what it looks like in the first image.

**Comment by davuxcom** [comment]
This is the intended behavior. Spanning across screens creates a different set of problems, so this was the solution. This window does not remember size/position across open/close either.

**Comment by MikeyMike22** [comment]
Why would that be intended? I did not mention spanning across screens, why would we want it to span across screens? @davuxcom

**Comment by davuxcom** [comment]
> Why would that be intended? I did not mention spanning across screens, why would we want it to span across screens? @davuxcom

This is why it doesn't just open up and show all your devices. It makes a cut-off at some point because we didn't want it crossing screen boundaries.  I think we pick something like no wider than 70% of the screen it'll open on, as the default width for the window.

**Reply by davuxcom** [reply]
I looked at this again. We pick a max of 3 devices and don't allow the default size to exceed that, no matter how big your screen is. Even if you had a large screen I don't think it's necessarily right to open it very large. So, it's actually not really clear what we should do. The feature you want is "restore window size" which is reasonable. The reason that this doesn't restore is that there's some additional code necessary to reposition windows back onscreen when the display they are on becomes unavailable, or the display configuration changes between sessions of the application.

So this is by design but it's agreed that the design isn't ideal. It may sound simple to reposition the window for these scenarios but I chose the version with fewer bugs over the more complicated better solution.

**Reply by MikeyMike22** [reply]
Ok, I understand that you chose that version, but the feature request remains, we would appreciate to have the better solution.
